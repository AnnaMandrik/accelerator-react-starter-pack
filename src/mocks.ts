import {Guitars} from './types/guitar';
import {image, music} from 'faker';

enum Count {
  GuitarsTest = 5,
  CommentsTest = 15,
}

export enum HttpCode {
  Ok = 200,
  NoContent = 204,
  NoAuth = 401,
}

export const makeFakeGuitar = () => ({
  'id': 1,
  'name': 'Честер Plus',
  'vendorCode': 'SO7568698',
  'type': music.genre(),
  'description': 'Вариант для настоящих профессионалов. Двенадцатиструнный инструмент оснащён карбоновыми струнами и корпусом из массива ели.',
  'previewImg': image.image(),
  'stringCount': 12,
  'rating': 5,
  'price': 1500,
});

export const makeFakeGuitars = (): Guitars => new Array(Count.GuitarsTest).fill(undefined).map(() => makeFakeGuitar());
