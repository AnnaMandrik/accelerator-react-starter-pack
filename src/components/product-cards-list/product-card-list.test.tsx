import { render } from '@testing-library/react';
import { Router} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import ProductCardsList from './product-cards-list';
import {makeFakeGuitars} from '../../mocks';

const history = createMemoryHistory();
const fakeGuitars = makeFakeGuitars();

describe('Component: ProductCardsList', () => {
  it('should render correctly', () => {
    const {container} = render(
      <Router history={history}>
        <ProductCardsList productsList={fakeGuitars}/>
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
