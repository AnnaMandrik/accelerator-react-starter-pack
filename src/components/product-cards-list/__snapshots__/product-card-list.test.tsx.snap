// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductCardsList should render correctly 1`] = `
<div>
  <div
    class="cards catalog__cards"
  >
    <div
      class="product-card"
    >
      <img
        alt="Честер Plus"
        height="190"
        src="http://placeimg/content.com/640/480/cats"
        width="75"
      />
      <div
        class="product-card__info"
      >
        <div
          aria-hidden="true"
          class="rate product-card__rate"
        >
          <span
            class="visually-hidden"
          >
            Рейтинг:
          </span>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <span
            class="rate__count"
          >
            5
          </span>
          <span
            class="rate__message"
          />
        </div>
        <p
          class="product-card__title"
        >
          Честер Plus
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          1500
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <a
          class="button button--mini"
          data-testid="more"
          href="#"
        >
          Подробнее
        </a>
        <a
          class="button button--red button--mini button--add-to-cart"
          href="#"
        >
          Купить
        </a>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt="Честер Plus"
        height="190"
        src="http://placeimg/content.com/640/480/fashion"
        width="75"
      />
      <div
        class="product-card__info"
      >
        <div
          aria-hidden="true"
          class="rate product-card__rate"
        >
          <span
            class="visually-hidden"
          >
            Рейтинг:
          </span>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <span
            class="rate__count"
          >
            5
          </span>
          <span
            class="rate__message"
          />
        </div>
        <p
          class="product-card__title"
        >
          Честер Plus
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          1500
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <a
          class="button button--mini"
          data-testid="more"
          href="#"
        >
          Подробнее
        </a>
        <a
          class="button button--red button--mini button--add-to-cart"
          href="#"
        >
          Купить
        </a>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt="Честер Plus"
        height="190"
        src="http://placeimg/content.com/640/480/fashion"
        width="75"
      />
      <div
        class="product-card__info"
      >
        <div
          aria-hidden="true"
          class="rate product-card__rate"
        >
          <span
            class="visually-hidden"
          >
            Рейтинг:
          </span>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <span
            class="rate__count"
          >
            5
          </span>
          <span
            class="rate__message"
          />
        </div>
        <p
          class="product-card__title"
        >
          Честер Plus
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          1500
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <a
          class="button button--mini"
          data-testid="more"
          href="#"
        >
          Подробнее
        </a>
        <a
          class="button button--red button--mini button--add-to-cart"
          href="#"
        >
          Купить
        </a>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt="Честер Plus"
        height="190"
        src="http://placeimg/content.com/640/480/sports"
        width="75"
      />
      <div
        class="product-card__info"
      >
        <div
          aria-hidden="true"
          class="rate product-card__rate"
        >
          <span
            class="visually-hidden"
          >
            Рейтинг:
          </span>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <span
            class="rate__count"
          >
            5
          </span>
          <span
            class="rate__message"
          />
        </div>
        <p
          class="product-card__title"
        >
          Честер Plus
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          1500
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <a
          class="button button--mini"
          data-testid="more"
          href="#"
        >
          Подробнее
        </a>
        <a
          class="button button--red button--mini button--add-to-cart"
          href="#"
        >
          Купить
        </a>
      </div>
    </div>
    <div
      class="product-card"
    >
      <img
        alt="Честер Plus"
        height="190"
        src="http://placeimg/content.com/640/480/cats"
        width="75"
      />
      <div
        class="product-card__info"
      >
        <div
          aria-hidden="true"
          class="rate product-card__rate"
        >
          <span
            class="visually-hidden"
          >
            Рейтинг:
          </span>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="11"
            width="12"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <span
            class="rate__count"
          >
            5
          </span>
          <span
            class="rate__message"
          />
        </div>
        <p
          class="product-card__title"
        >
          Честер Plus
        </p>
        <p
          class="product-card__price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          1500
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <a
          class="button button--mini"
          data-testid="more"
          href="#"
        >
          Подробнее
        </a>
        <a
          class="button button--red button--mini button--add-to-cart"
          href="#"
        >
          Купить
        </a>
      </div>
    </div>
  </div>
</div>
`;
